changeset/lib/rom/changeset/update.rb

```diff
@@ -65,7 +65,7 @@ def clean?
       def diff
         @diff ||=
           begin
-            source = Hash(original)
+            source = original.to_h
```

Comment:
@flash-gordon `Hash` expects `to_h` so this is fine BUT now you'd get `NoMethodError` rather than `TypeError` so we should probably maintain this behavior

Reply:
@solnic but what's the point? `.to_h` is a less strict constraint, you almost certainly won't get an error during upgrade. Defensive programming doesn't help unless you expect something to fail. And, finally, differentiating between `NoMethodError` and `TypeError` is a joke in a language like ruby.

Reply:
OK fair enough

Reply:
@solnic to be completely fair, I'm in favor of adding semantics whenever possible. This for sure does include robust using exception types. But when we don't expect a failure it's better not to bother